### ISSUE TITLE
A question about luarocks remove and luarocks purge (feature request?)

### ISSUE THREAD

**Issue by telemachus** [luarocks/luarocks#750]
os: macOS 10.12.6
lua: Lua 5.3.4
luarocks: 2.4.3

I'm wondering about the design of `luarocks remove` (and `purge` as well), and in particular I wonder if there's room (or need) for a new command or a new flag for `remove` and `purge`. For the moment, I'll call this flag `--extra-force`—though obviously that's a pretty poor name.

The need I see is this: if a user's installation of luarocks gets into a confused enough state, `luarocks remove` simply gives up, even if you add the existing `--force` flag. The user is left with nothing to do but manually `rm -rf` directories.

I just had this experience. Homebrew has been switching between different versions of both lua and luarocks as their default, and somehow my installation ended up a mess. `luarocks list` listed several rocks that were not present at all or only partially present, but when I tried to remove them, even using `--force`, luarocks would tell me "Could not remove \<so-and-so\>: \<so-and-so\> not found". I tried using `luarocks purge`, and ran into the same problem.

In one or two cases, I was able to use `touch` to fool luarocks into removing messed up installations, but that was too tedious when the rock installed 20-30 small files, so I gave up and resorted to `rm -rf`.

tl;dr - Is there space for a luarocks command or flag for a user who has a broken installation and still wants to remove a rock (to whatever degree is possible) and start fresh?

**Comment by mpeterv** [comment]
I don't remember if I made the checks that strict in #631 and #644 or if they were like that before, but now I think that the behaviour you want should be the default. As in, there should be a warning when trying to delete a file that doesn't exist already, but no error. It's just not helpful when trying to repair broken installations and it doesn't introduce any new breakage.

**Comment by telemachus** [comment]
@mpeterv Thanks for the reply. I agree that the less strict behavior (with a warning) would be a better default. I'd love to see this change in a future version of luarocks. Thanks again.

**Comment by hishamhm** [comment]
@mpeterv Indeed, the differentiation between "not found" and "fail" was lost along the way. I just put it back in eb89bafa29cf63600154d162429c803bb782c428 -- I believe this fixes the issue?

**Comment by telemachus** [comment]
Thanks to both of you!
